vertical integration

BotrosHanna-INL opened the pull request.

dylanjm commented on vertical_integration/src/convert_utils.py:
```diff
@@ -0,0 +1,192 @@
+import sys
+import shutil
+
+def createBackup(filename):
```
stay consistent with snake_case or camelCase.

dylanjm commented on vertical_integration/src/aspen_to_heron.py:
```diff
@@ -0,0 +1,66 @@
+"""
```
I think the conversion scripts should all be moved to a `vertical_integration/scripts` folder and the function that do the conversion should be moved into the respective script from `main.py`

Also, all scripts that will be run by a user in the command line should start with a `#!/usr/bin/env python` line

BotrosHanna-INL replied:
Done

dylanjm commented on vertical_integration/FORCE_Components/ComponentSetsFiles/Sets1/ReadMe:
```diff
@@ -0,0 +1,36 @@
+
+Example:
+
+{
```
It would be best if you made the examples render as code  blocks you can do that by surrounding the chunk in three backticks (```)

BotrosHanna-INL replied:
Done

BotrosHanna-INL commented on vertical_integration/src/main.py:
```diff
@@ -0,0 +1,661 @@
+The objective of this code is the vertical integration (auomated data transfer) between different IES codes. To faciliate this integration, vaious classes and methods are created. This code includes the following:
+1 - Python Classes for the component info in Aspen HYSYS, and Aspen APEA
+2 - A Python Class for the "FORCE Component". "FORCE Component" is a component that combines the component's inforation from two (or more) codes.
+3 - A Python Class for the "FORCE Component Set". The "FORCE Component Set" is set of components grouped together. For example: grouping all the pumps together. The component set is created to produce the cost function of a specific component category (e.g. a pump or a turbine). It is also needed to create a component set to be used in HERON.
```
done

BotrosHanna-INL commented on vertical_integration/src/main.py:
```diff
@@ -0,0 +1,661 @@
+  """
+    The APEA component: the APEA component is defined by the component name and the APEA output xlsx file from which the component info is imported (it is possible to have two compoennts with the same name) from different xlsx APEA outputs
+  """
+  def __init__(self,xlsx_filename,component_name):
```
done

BotrosHanna-INL commented on vertical_integration/src/main.py:
```diff
@@ -0,0 +1,661 @@
+    equipment_weight  = file_data.loc[file_data['Name'] == self.component_name, 'Equipment Weight [LBS]'].values[0]
+    total_installed_weight = file_data.loc[file_data['Name'] == self.component_name, 'Total Installed Weight [LBS]'].values[0]
+
+    apea_comp_info= {"APEA Component Name": self.component_name, "APEA Source":self.xlsx_filename,"APEA Equipment Cost [USD]": equipment_cost.item(),"APEA Installed Cost [USD]":installed_cost.item() , "APEA Equipment Weight [LBS]": equipment_weight.item(), "APEA Total Installed Weight [LBS]": total_installed_weight.item()}
```
done

BotrosHanna-INL commented on vertical_integration/src/main.py:
```diff
@@ -0,0 +1,661 @@
+# Section 4:
+# A Method to create/update a component (or a component set) in HERON using the components' info from Aspen HYSYS and APEA
+
+def create_componentSets_in_HERON(comp_sets_folder, heron_input_xml):
```
changed it

BotrosHanna-INL commented on vertical_integration/src/create_all_forceComponents_from_aspen_apea.py:
```diff
@@ -0,0 +1,25 @@
+  parser.add_argument("folder2_HYSES_or_APEA", help="folder2_HYSES_or_APEA")
+  args = parser.parse_args()
+
+create_all_force_components_from_hysys_apea(args.folder1_HYSES_or_APEA, args.folder2_HYSES_or_APEA)
```
I will keep the scripts that the user will use short and keep the big classes and methods on the src folder.
I added the copyright notice.

BotrosHanna-INL commented:
@dylanjm 
Thank you so much, Dylan for your comments. I have addressed them. Please feel free to add more comments if needed.

dylanjm reviewed (changes requested):
A few things we should consider:

- I don't think we should break code into `main_classes` vs. `main_methods` I think we should segment code by their application. So all Aspen and AspenHysys methods and classes go into an APEA class. 
- We should be able to add `__init__.py` files to make the importing of methods into our scripts more clear. (i.e. line 17 `scripts/create_apea_components.py`)
- The scripts still call the important function outside the scope of main computation (i.e. line 29 `scripts/create_apea_components.py` should be indented one level)
- How many of the excel files do we need to demonstrate how this vertical integration works? Where do we draw the line for examples and integration testing?

BotrosHanna-INL commented:
- For a class like "APEA component", I have some methods already but the "main_methods" are methods that goes over many components. I do not think they should go under the "APEA_component" class
-  I added the __init__.py file.
- I modified the scripts as you requested.
- Yes, I have several excel files but actually all of them were used for the TES case. In other words, it is one test because one HERON component can be equivalent to several components and the creating a cost function for one component requires cost data from several excel files. Since running that one test (aspen to Heron) is not computationally expensive, I think it is OK to leave all the excel files there.
